### PR TITLE
Reset password

### PR DESCRIPTION
**Pull request by nichitanovicov**

### CODE REVIEW

File: IdentityService.Tests/Endpoints/User/ResetPasswordEndpointHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+using IdentityService.Tests.Helpers;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using Moq;
+
+namespace IdentityService.Tests.Endpoints.User
+{
```
Comment (ghost) [inline IdentityService.Tests/Endpoints/User/ResetPasswordEndpointHandlerTests.cs:14]: remove the curly brackets.

File: IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using IdentityService.Models;
+using Microsoft.AspNetCore.Identity;
+
+namespace IdentityService.Endpoints.User
+{
```
Comment (ghost) [inline IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs]: you can remove brackets here.

File: IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    public class ResetPasswordEnpointHandler
+    {
+        public static async Task<IResult> ResetPassword(ResetPassword model, UserManager<ApplicationUser> userManager, CancellationToken cancellationToken) 
+        {
+            ApplicationUser? user = await userManager.FindByEmailAsync(model.Email);
```
Comment (ghost) [inline IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs]: you can use just `var user`

File: IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            {
+                return Results.BadRequest(new List<IdentityError> {
+                    new IdentityError
+                {
+                    Code = "Password",
+                    Description = "The password and confirmation password do not match."
+                }});
+            }
+
+            IdentityResult result = await userManager.ResetPasswordAsync(user, model.Code, model.Password);
```
Comment (ghost) [inline IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs]: you can use `var result`

Comment (nichitanovicov) [reply IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs]: I think using type names will increases readability of code

File: IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs (hunk 2 above)
Comment (ghost) [reply IdentityService/Endpoints/User/ResetPasswordEnpointHandler.cs]: this is not resolved yet.

File: IdentityService.Tests/Endpoints/User/ResetPasswordEndpointHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            Password = "passwordddd",
+            ConfirmPassword = "passwordddd",
+        };
+        _currentUser = new ApplicationUser();
+
+        _resetPasswordAsyncSetup = _userManagerMock.Setup(c => c.ResetPasswordAsync(_currentUser, _defaultModel.Code, _defaultModel.Password));
+        _findByEmailAsyncSetup = _userManagerMock.Setup(c => c.FindByEmailAsync(_defaultModel.Email));
+    }
+    [Fact]
+    public async Task ResetPassword_WhenUserUserWithThisEmailDoesNotExists_ReturnsBadRequestWithIdentityError()
```
Comment (ddimma) [inline IdentityService.Tests/Endpoints/User/ResetPasswordEndpointHandlerTests.cs]: Must be a typo UserUser?

**Comment by ghost** [comment]
n-am cum sa dau approve, dar e approved din partea mea :)

**Review by balabanvladislav — APPROVED** [review]

**Review by ddimma — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by DaniellaEndava — APPROVED** [review]

**Pull request merged**